h2/src/main/org/h2/expression/ConditionAndOr.java

```diff
@@ -300,4 +307,40 @@ public Expression getSubexpression(int index) {
         }
     }
 
+    /**
+     * Optimize query according to the given condition. Example:
+     * (A AND B) OR (C AND B), the new condition (A OR C) AND B is returned
+     *
+     * @param left the session
+     * @param right the second condition
+     * @return null or the third condition
+     */
+    private Expression optimizeConditionAndOr(ConditionAndOr left, ConditionAndOr right){
+        if(left.andOrType == AND && right.andOrType == AND){
+            Expression commonExpression;
+            Expression combinedExpression;
+            if(left.getSubexpression(0).getSQL().equals(right.getSubexpression(0).getSQL())){
```

Comment:
left.getSubexpression(0).getSQL() and right.getSubexpression(0).getSQL() should be extracted into variables, so we only need to call them once

Reply:
Resolved in the last commit